Formula/crunchy-cli.rb

```diff
@@ -0,0 +1,28 @@
+class CrunchyCli < Formula
+  desc "Command-line downloader for Crunchyroll"
+  homepage "https://github.com/crunchy-labs/crunchy-cli"
+  url "https://github.com/crunchy-labs/crunchy-cli/archive/refs/tags/v3.0.0.tar.gz"
+  sha256 "7a693668bff8aca9029106bb143692cf74eb2376b568aea8173e4b6af45c0b45"
+  license "MIT"
+  head "https://github.com/crunchy-labs/crunchy-cli.git", branch: "master"
+
+  depends_on "rust" => :build
+  depends_on "ffmpeg"
+  depends_on "openssl@3"
+
+  on_linux do
+    depends_on "pkg-config" => :build
+  end
+
+  def install
+    system "cargo", "install", "--no-default-features", "--features", "openssl", *std_cargo_args
+    man1.install Dir["target/release/manpages/*"]
+    bash_completion.install "target/release/completions/crunchy-cli.bash"
+    fish_completion.install "target/release/completions/crunchy-cli.fish"
+    zsh_completion.install "target/release/completions/_crunchy-cli"
+  end
+
+  test do
+    assert_match "crunchy-cli #{version}", shell_output("#{bin}/crunchy-cli --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I've added a test to check for invalid credentials as every command needs the user to be logged-in order to work